ucode.c

```diff
@@ -79,22 +80,32 @@ ucode_load(const char *file) {
 	return progfunc;
 }
 
+static void
+uc_uloop_timeout_free(struct ucrun_timeout *timeout)
+{
+	ucv_put(timeout->function);
+	ucv_put(timeout->priv);
+	list_del(&timeout->list);
+	free(timeout);
+}
+
 static void
 uc_uloop_timeout_cb(struct uloop_timeout *t)
 {
-	struct urun_timeout *timeout = container_of(t, struct urun_timeout, timeout);
+	struct ucrun_timeout *timeout = container_of(t, struct ucrun_timeout, timeout);
 	uc_value_t *retval = NULL;
 
-	/* push the function to the stack */
-	uc_vm_stack_push(&timeout->urun->vm, ucv_get(timeout->function));
-
+	/* push the function and private data to the stack */
+	uc_vm_stack_push(&timeout->ucrun->vm, ucv_get(timeout->function));
+	if (timeout-> priv)
+		uc_vm_stack_push(&timeout->ucrun->vm, ucv_get(timeout->priv));
 	/* invoke function with zero arguments */
-	if (uc_vm_call(&timeout->urun->vm, false, 0)) {
+	if (uc_vm_call(&timeout->ucrun->vm, false, timeout->priv ? 1 : 0)) {
```

Comment:
… ant the ternary expression here. Simply always pass ->priv. If it is unset, the ucode function will receive null as argument